tests/README.md

```diff
@@ -0,0 +1,293 @@
+
+Instead of writing result cases in tests and updating them when behaviour changes, a snapshot-based testing workflow relies on auto-generation. Whenever a `.new` snapshot is generated, it means one of the following:
+
+- Your code made an unexpected breaking change, and you need to fix it.
+- Your code made an expected breaking change, and you need to accept the new snapshot.
+
+It is your job to determine which one is the case, and take action accordingly.
+
+I recommend using `cargo-insta`, especially this command:
+
+```bash
+cargo insta test --review
+```
+
+This will regenerate all snapshots without interrupting the test every time there's a diff, and it will also open the snapshot review interface, so you can accept or reject `.new` snapshots.
+
+### `http_spec`
+
+1. The name, and optionally the description, are read out and put into a [header](#header).
+1. The GraphQL config is read out and put into a [`server` block](#server).
+1. The client schema snapshot is auto-generated for the [`server` block](#server).
+1. The merged SDL snapshot is auto-generated for the [`server` block](#server).
+1. The `assert` property's `response` are removed from the YAML and converted into snapshots.
+1. The `assert` property is converted to an [`assert` block](#assert).
+1. The `mock` property is put into a [`mock` block](#mock).
+1. If the `http_spec` had a `runner: fail` annotation (which is unsupported by `execution_spec`), the snapshot of the error is automatically generated.
+
+### `graphql_spec`
+
+#### client (1 server-sdl, 1 client-sdl)
+
+1. The name is generated from the file name, and put into a [header](#header).
+1. The server SDL is put into a [`server` block](#server).
+1. The client SDL is put into the corresponding `client` snapshot.
+1. The merged SDL snapshot is auto-generated for the [`server` block](#server).
+
+These tests will not have a main action, since `client` checks are performed on all (non-`sdl error`) tests with only one [`server` block](#server).
+
+#### merge (1+ server-sdl, 1 merged-sdl)
+
+1. The name is generated from the file name, and put into a [header](#header).
+1. The server SDLs are put into [`server` blocks](#server).
+1. The merged SDL is put into the a `merged` snapshot.
+1. If the test only has one server block, the client schema snapshot is auto-generated for the [`server` block](#server).
+
+#### errors (1 server-sdl, 1 client-sdl with `@error` operator)
+
+Each test's `.md` file may have a file name suffix of `-error` if the bare file name is already taken by a previously converted test.
+
+1. The name is generated from the file name, and put into a [header](#header).
+1. An [`sdl error` instruction](#instruction) is appended.
+1. The server SDL is put into a [`server` block](#server).
+1. The expected errors are put into an `errors` snapshot.
+
+## Maintenance
+
+1. To clean unused snapshots, run `cargo insta test --delete-unreferenced-snapshots`.
```

Comment:
We don't need this file. Changes should be incorporated https://github.com/tailcallhq/tailcallhq.github.io/pull/208